spinalcordtoolbox/reports/qc.py

```diff
@@ -223,11 +229,26 @@ def _make_QC_image_for_3d_volumes(self, img, mask, plane, imgs_to_generate):
         fig = mpl_figure.Figure()
         fig.set_size_inches(size_fig[0], size_fig[1], forward=True)
         mpl_backend_agg.FigureCanvasAgg(fig)
-        ax = fig.add_axes((0, 0, 1, 1))
-        ax.imshow(img, cmap='gray', interpolation=self.interpolation, aspect=float(self.aspect_img))
-        self._add_orientation_label(ax)
-        ax.get_xaxis().set_visible(False)
-        ax.get_yaxis().set_visible(False)
+        if self.process == 'sct_fmri_compute_tsnr':
+            colorbar = True
+            ax = fig.add_axes((0, 0, 0.93, 1))
+            cmap = 'seismic'
+            vmin = min(int(np.amin(img)), int(np.amin(mask)))
+            vmax = max(int(np.amax(img)), int(np.amax(mask[0]))) - 2
+            norm = mpl_colors.Normalize(vmin=vmin, vmax=vmax)
+        else:
+            ax = fig.add_axes((0, 0, 1, 1))
+            cmap = 'gray'
+            norm = None
+            colorbar = False
```

Comment:
I think we might be able to simplify these default values a little more by instead making them the default values inside of `no_seg_seg`:

```python
    def no_seg_seg(self, mask, ax, cmap='gray', norm=None, fig=None, colorbar=False):
```

By using keywords, we can then construct a dictionary for the keyword arguments:

```python
kwargs = {}
if self.process == 'sct_fmri_compute_tsnr': 
    kwargs['cmap'] = 'seismic'
    kwargs['colorbar'] = True
    kwargs['fig'] = fig
    vmin = min(int(np.amin(img)), int(np.amin(mask)))
    vmax = max(int(np.amax(img)), int(np.amax(mask[0]))) - 2
    kwargs['norm'] = mpl_colors.Normalize(vmin=vmin, vmax=vmax)
```

Which lets us use a single `action()` function signature as before, by using [`**kwargs`](https://www.digitalocean.com/community/tutorials/how-to-use-args-and-kwargs-in-python-3#using-args-and-kwargs-in-function-calls) expansion to fill in the remaining (optional) parameters:

```python
   # here
   QcImage.no_seg_seg(self, mask, ax, **kwargs)
   
   # later on
   action(self, mask[i], ax, **kwargs)
```

(Saving us the need for an additional `if/else` block when calling `action`.)

Reply:
Thanks for the suggetsion! Now done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4602/commits/6fd1a2f9d231394d4611533cfd8f4d49a721e461